lib/streams/compound-duplex.js

```diff
@@ -63,12 +63,40 @@ function CompoundDuplex(writable, readable, options) {
 		self.push(null);
 	});
 
-	this.on('finish', function() {
-		writable.end();
+	readable.on('finish', function() {
```

Comment:
Manually pushing the EOF is what also happens after `Transform._flush`.

I'll play around with it a bit more to see if I can get away with not manually emitting a `finish` event.